support-frontend/assets/pages/[countryGroupId]/components/checkoutComponent.tsx

```diff
@@ -134,8 +134,42 @@ function paymentMethodIsActive(paymentMethod: LegacyPaymentMethod) {
 	);
 }
 
+// ----- Setup ----- //
+const POLLING_INTERVAL = 3000;
+const MAX_POLLS = 10;
+
+// retry mechanism for polling based upon
+// https://bpaulino.com/entries/retrying-api-calls-with-exponential-backoff
+function timeOut(milliseconds: number | undefined): Promise<void> {
+	return new Promise((resolve) => setTimeout(resolve, milliseconds));
+}
+function retry(
```

Comment:
Since this retry code is very generic I wonder if it should be moved to its own module? We could also write some tests for if we pulled it out. I think the only thing which ties it concretely to our use case is the `Promise<Response>` in the args, but if we changed this use TS generics, it would work for any use case (and would make the testing a bit simpler):

```
function retry<A>(
    promise: Promise<A>,
...
```
```

Reply:
Agree, we have done this and tests now in files `retryPaymentStatus.ts` and `retryPaymentStatusTest.ts`